src/main/java/com/mixpanel/android/viewcrawler/ViewVisitor.java

```diff
@@ -99,12 +103,66 @@ protected String name() {
             return "Property Mutator";
         }
 
-        private final Caller mMutator;
-        private final Caller mAccessor;
+        private final PropertyCaller mMutator;
+        private final PropertyCaller mAccessor;
         private final WeakHashMap<View, Object> mOriginalValues;
         private final Object[] mOriginalValueHolder;
     }
 
+    public static class LayoutSetVisitor extends ViewVisitor {
+        public LayoutSetVisitor(List<Pathfinder.PathElement> path, JSONObject args) {
+            super(path);
+            mOriginalValues = new WeakHashMap<View, JSONObject>();
+            mArgs = args;
+        }
+
+        @Override
+        public void cleanup() {
+            for (Map.Entry<View, JSONObject> original:mOriginalValues.entrySet()) {
+                final View changedView = original.getKey();
+                final JSONObject originalValue = original.getValue();
+                setLayout(changedView, originalValue.optInt("verb"), originalValue.optInt("anchor"));
+            }
+        }
+
+        @Override
+        public void accumulate(View found) {
+            final int newRuleIndex = mArgs.optInt("verb");
+            final int newAnchorId = mArgs.optInt("anchor");
+            final RelativeLayout.LayoutParams currentParams = (RelativeLayout.LayoutParams)found.getLayoutParams();
+            final int[] currentRules = currentParams.getRules();
+
+            if (currentRules[newRuleIndex] == newAnchorId) {
+                return;
+            }
+
+            if (mOriginalValues.containsKey(found)) {
+                ; // Cache exactly one
+            } else {
+                JSONObject originalValue = new JSONObject();
+                try {
+                    originalValue.put("verb", newRuleIndex);
+                    originalValue.put("anchor", currentRules[newRuleIndex]);
+                } catch (JSONException e) {
+                    ; // keys won't be null..
+                }
+                mOriginalValues.put(found, originalValue);
+            }
+            setLayout(found, newRuleIndex, newAnchorId);
+        }
+
+        private void setLayout(View target, int ruleIndex, int anchorId) {
+            RelativeLayout.LayoutParams params = (RelativeLayout.LayoutParams)target.getLayoutParams();
+            params.addRule(ruleIndex, anchorId);
+            target.setLayoutParams(params);
+        }
+
+        protected String name() { return "Layout Mutator"; }
```

Comment:
ah.. I saw this in property visitor:
```
protected String name() {
            return "Property Mutator";
        }
```

What should be the proper return value in this case?